### ISSUE TITLE
ALOpsAppSign with 'Hosted Agent' -> msvcr120 error?

### ISSUE THREAD

**Issue by skonsbruck** [HodorNV/ALOps#463]
Good morning.

Anyone tried to use a multistage-pipeline in combination with a Microsoft 'Hosted Agent' and ALOpsAppSign?

We receive the following error message: 
##[error]The process cannot access the file 'C:\Windows\System32\msvcr120.dll' because it is being used by another process.

Here are the details of the 'Hosted Agent' (windows-latest)
Starting: Initialize job
Agent name: 'Hosted Agent'
Agent machine name: 'WIN-MNCHU94BDNL'
Current agent version: '2.195.1'
Operating System
Microsoft Windows Server 2019
10.0.17763
Datacenter
Virtual Environment
Environment: windows-2019
Version: 20211229.2
Included Software: https://github.com/actions/virtual-environments/blob/win19/20211229.2/images/win/Windows2019-Readme.md
Image Release: https://github.com/actions/virtual-environments/releases/tag/win19%2F20211229.2

Any input would be much appreciated. 
Best regards.

**Comment by waldo1001** [comment]
We were able to repro the issue, and we have a hunch what it causes.  We'll try to implement a fix.

**Comment by waldo1001** [comment]
This should be fixed.  Can you confirm?  Could be you have to run the pipeline twice to make sure the new version is downloaded.

**Comment by skonsbruck** [comment]
Hello, thanks for your feedback and resolution attempt!

I tried to run the pipeline twice, on the first attempt the error was ##[error]Signature status: UnknownError. The form specified for the subject is not one supported or known by the specified trust provider.

On the second attempt it was the same ##[error]Signature status: UnknownError. The form specified for the subject is not one supported or known by the specified trust provider.

Now, maybe this is caused by a different issue... I have to investigate on our end as well...

**Comment by waldo1001** [comment]
Yeah, make sure the certificate is ok.  And - if you have the opportunity to try - try it out on a self-hosted server?

**Comment by nixpt82** [comment]
@waldo1001 

Yes, just tried in a Self-hosted server and validates the certificate

The error bellow is when i run the pipeline in MS-Hosted in debug.

![image](https://user-images.githubusercontent.com/35833855/145796140-3173767e-e721-425a-a9b4-997c3e46c6ac.png)

**Comment by waldo1001** [comment]
can you try without debug mode?

**Comment by nixpt82** [comment]
Yes, got the same issue

![image](https://user-images.githubusercontent.com/35833855/145819222-9049b18e-73ad-477a-9073-734d308dddbf.png)

**Comment by waldo1001** [comment]
I think I found the issue .. seems the timestamp-url isn't always stable.

We seem to have good results with:

```yaml
  - task: ALOpsAppSign@1
    inputs:
      pfx_path: $(pfxurl)
      pfx_password: $(pfxpwd)
      timestamp_uri: "http://timestamp.digicert.com/"
```

**Comment by nixpt82** [comment]
Still got the same error but by doing the "verify sign" task, i got this: 

`##[warning]App is Signed but not timestamped`

![image](https://user-images.githubusercontent.com/35833855/146386807-a7eceb0e-d7f9-491a-a5c8-0fe261f34a0a.png)

Maybe a workaround:

[https://docs.microsoft.com/en-us/windows/msix/desktop/cicd-keyvault](url)

**Comment by waldo1001** [comment]
Yeah, it worked once for me, and now not anymore - only on MS Hosted crap (sorry).

Would you be able to use your own build machine for it?

**Comment by nixpt82** [comment]
yes, the self hosted machines never got this issue. I'm sure it runs on it, I just want to scrap my last VM on azure.I will give a try in the azure sign option and give you feedback if it works, maybe you can make a task for it.I bit of fan of having all the things in Key Vault. I have all dev running on the azure containers instances, what is bit cheaper then azure VMs
-------- Original message --------From: waldo ***@***.***> Date: 12/16/21  18:32  (GMT+01:00) To: HodorNV/ALOps ***@***.***> Cc: LM ***@***.***>, Comment ***@***.***> Subject: Re: [HodorNV/ALOps] ALOpsAppSign with 'Hosted Agent' -> msvcr120 error? (Issue #463) 
Yeah, it worked once for me, and now not anymore - only on MS Hosted crap (sorry).
Would you be able to use your own build machine for it?

**Comment by nixpt82** [comment]
> 

I test with the Azure Sign Tool but seems the tool cannot be use for *.app files.  :( 
Im bank to the old way, VM with a Self-Hosted agent to sign the apps.

**Comment by waldo1001** [comment]
Sorry - we're highly dependent on the signing-tools, I'm afraid :(
